libyul/YulString.h

```diff
@@ -76,12 +83,18 @@ class YulStringRepository: boost::noncopyable
 			hash *= 1099511628211u;
 			hash ^= c;
 		}
+		for (int i = 0; i < 8; i++)
+		{
+			hash *= 1099511628211u;
+			hash ^= s & 0xFF;
+            s = s >> (56 - i);
```

Comment:
This will already shift right ``56`` bits the first time and will then ``s`` will always be zero starting from ``i = 2``. You probably won't notice this soon, but this will produce the same hashes starting from a suffix of 256. I think you meant to shift right 8 bits every iteration.
```suggestion
			s >>= 8;
```

Actually, we could also change the ``for(...)`` into a ``while(s)`` and stop hashing in zeros in the end.

Reply:
Yup--the reason for `(56-i)` was for the previous way using a byte-array. I'll use your suggestion.